internal/utils.go

```diff
@@ -42,14 +45,49 @@ func filterByRegex(pattern string, fi []*funcInfo) ([]*funcInfo, error) {
 
 func getFilename(filePath string) (string, error) {
 	dir, file := filepath.Split(filePath)
-	pkg, err := build.Import(dir, ".", build.FindOnly)
-	if err != nil {
-		return "", err
-	}
+	// TODO: refactor this to simpler approach
+	var pkg *build.Package
+	var err error
+	fext := strings.Split(file, ".")[1]
+	if strings.EqualFold(fext, "go") {
+		pkg, err = build.Import(strings.TrimSuffix(dir, "/"), ".", build.FindOnly)
+		if err != nil {
+			if strings.Contains(err.Error(), "go get") {
+				if err := importModule(strings.TrimSuffix(dir, "/")); err != nil {
+					return "", err
+				}
+			}
+			return "", err
+		}
+
+	} else {
+		pkg, err = build.ImportDir(dir, build.FindOnly)
+		if err != nil {
+			_, err = build.Import(dir, ".", build.FindOnly)
+			if err != nil {
+				if strings.Contains(err.Error(), "add") {
+					if err := importModule(strings.TrimSuffix(dir, "/")); err != nil {
+						return "", err
+					}
```

Comment:
Could you please explain these in simpler terms? How does the error checking happen here?

Reply:
added comments https://github.com/burntcarrot/samosa/pull/4/commits/31af7604145dd5a0bc3d1b54ddeab41b48a9394d commit as per request